chain/messagepool/messagepool.go

```diff
@@ -1001,19 +1022,19 @@ func (mp *MessagePool) addLocked(ctx context.Context, m *types.SignedMessage, st
 }
 
 func (mp *MessagePool) GetNonce(ctx context.Context, addr address.Address, _ types.TipSetKey) (uint64, error) {
-	mp.curTsLk.Lock()
-	defer mp.curTsLk.Unlock()
+	mp.curTsLk.RLock()
+	defer mp.curTsLk.RUnlock()
 
-	mp.lk.Lock()
-	defer mp.lk.Unlock()
+	mp.lk.RLock()
+	defer mp.lk.RUnlock()
 
 	return mp.getNonceLocked(ctx, addr, mp.curTs)
```

Comment:
This currently needs a write lock for:

https://github.com/filecoin-project/lotus/blob/841b7b873005c6ddfb536125ca9a90b92b23d8c9/chain/messagepool/messagepool.go#L489-L490

We should consider replacing the `keyCache` with an LRU anyways? That locks internally so we wouldn't need to worry about this.